### PR TITLE
Upgrade to Bicleaner 3

### LINKED ISSUE #528: Upgrade to BicleanerAI 3.0
https://github.com/bitextor/bicleaner-ai/blob/v3.0.0/CHANGELOG.md

The release includes new multilingual models. We can try using `large` ones. see https://github.com/bitextor/bicleaner-ai/blob/v3.0.0/docs/training/multilingual.md#performance

### PR DESCRIPTION
**Pull request by eu9ene**
bicleaner2 full multilingual model: Troughput: 271 rows/s
bicleaner3 full-large multilingual model: Troughput: 80 rows/s

I would still try the large model as we use those multilingual models for non-European languages where we have less data.

https://firefox-ci-tc.services.mozilla.com/tasks/Kjhc2ehkTi-06k-nr8lwrg/runs/0/logs/public/logs/live.log

fixes #528

### CODE REVIEW

**Comment by eu9ene** [comment]
Bicleaner now requires https://github.com/MSeal/cython_hunspell/tree/2.0.3 which builds hunspell which needs some system libs [installed](https://github.com/hunspell/hunspell?tab=readme-ov-file#dependencies). Since this is not a docker worker I can't add them in this PR. I don't think there's a way to use our prebuilt packages. Also it seems cyhunspell wants version 1.7 and we fetch 0.5.5. 

I was able to successfully install and use it on a Snakepit machine under conda env.

File: taskcluster/docker/toolchain-build/Dockerfile [hunk 1]
```diff
@@ -19,7 +19,11 @@ RUN apt-get update -qq \
                           tar \
                           libxml2 \
                           libhunspell-dev \
-                          bc \
+                          bc  \
+                          autoconf \
```
Comment (eu9ene) [inline taskcluster/docker/toolchain-build/Dockerfile:23]: @bhearsum these are the four packages required by hunspell that we need to add on the generic worker

Comment (eu9ene) [reply taskcluster/docker/toolchain-build/Dockerfile:23]: ok, we don't need it with the toolchain, but I'd still leave if for the future when we fully switch to Docker. Then we can remove the toolchain.

**Comment by eu9ene** [comment]
Bicleaner folks also recommend switching to the multilingual model even for the languages that have specialized models. We can run an experiment side by side but I'm hesitant to do it before the next big training.

**Comment by marco-c** [comment]
> Bicleaner folks also recommend switching to the multilingual model even for the languages that have specialized models. We can run an experiment side by side but I'm hesitant to do it before the next big training.

Where did you see that? Did they share numbers?

File: pipeline/bicleaner/download_pack.py [hunk 2]
```diff
@@ -27,10 +27,12 @@
...
 # If a new model is released and you want to invalidate Taskcluster caches,
 # change this file since it is a part of the cache digest
-# The last model was added to https://huggingface.co/bitextor on Aug 29, 2023
+# The last model was added to https://huggingface.co/bitextor on Mar 11, 2024
 def _run_download(src: str, trg: str, dir: str) -> subprocess.CompletedProcess:
+    # use large multilingual models
+    model_type = "full-large" if trg == "xx" else "full"
     return subprocess.run(
-        ["bicleaner-ai-download", trg, src, "full", dir], capture_output=True, check=False
+        ["bicleaner-ai-download", src, trg, model_type, dir], capture_output=True, check=False
```
Comment (bhearsum) [inline pipeline/bicleaner/download_pack.py:35]: Just to sanity check: bicleaner reversed the src/trg arg order?

File: pipeline/bicleaner/requirements/bicleaner-ai.in [hunk 3]
```diff
@@ -1 +1,3 @@
-bicleaner-ai==2.0
+# cython_hunspell doesn't have wheels for Python 3.10
```
Comment (bhearsum) [inline pipeline/bicleaner/requirements/bicleaner-ai.in:1]: This comment doesn't make as much sense as it did when we were pointing at a git revision. Perhaps something like: "Note: this package is installed manually before the rest of the requirements", with a pointer to the kind?

**Review by bhearsum — APPROVED** [review]

File: pipeline/bicleaner/download_pack.py (hunk 2 above)
Comment (eu9ene) [reply pipeline/bicleaner/download_pack.py:35]: It was reversed only here, in the model downloading. There's only one direction for the bilceaner models and we try to download both so it didn't matter here. The bicleaner step always looks at metadata.yaml in the model package to determine the direction, so it worked correctly.

**Comment by eu9ene** [comment]
> > Bicleaner folks also recommend switching to the multilingual model even for the languages that have specialized models. We can run an experiment side by side but I'm hesitant to do it before the next big training.
> 
> Where did you see that? Did they share numbers?

https://github.com/bitextor/bicleaner-ai/issues/31#issuecomment-2118343598

**Review by gregtatum — APPROVED** [review]
Thanks for the upgrade!

**Pull request merged**